packages/service/lib/dependencies.js

```diff
@@ -0,0 +1,116 @@
+'use strict'
+const { readFile } = require('node:fs/promises')
+const { resolve } = require('node:path')
+const { closest } = require('fastest-levenshtein')
+const Topo = require('@hapi/topo')
+const ConfigManager = require('@platformatic/config')
+const errors = require('./errors')
+
+function missingDependencyErrorMessage (clientName, service, services) {
+  const closestName = closest(clientName, [...services.keys()])
+  let errorMsg = `service '${service.id}' has unknown dependency: '${clientName}'.`
+  if (closestName) {
+    errorMsg += ` Did you mean '${closestName}'?`
+  }
+  return errorMsg
+}
+
+async function getClientId (service, client) {
+  try {
+    const clientPackageJsonPath = resolve(service.path, client.path, 'package.json')
+    const clientPackageJSON = JSON.parse(await readFile(clientPackageJsonPath, 'utf8'))
+    return clientPackageJSON.name ?? ''
+  } catch (err) {
+    if (client.url === undefined || client.name === undefined) {
+      throw err
+    }
+  }
+}
+
+async function parseConfigUrl (configManager, urlString) {
+  if (!urlString) {
+    return { url: null, envVar: null }
+  }
+
+  try {
+    const url = await configManager.replaceEnv(urlString)
+    return { url, envVar: null }
+  } catch (err) {
+    // The MissingValueError is an error coming from pupa
+    // https://github.com/sindresorhus/pupa#missingvalueerror
+    // All other errors are simply re-thrown.
+    if (err.name !== 'MissingValueError' || urlString !== `{${err.key}}`) {
+      throw err
+    }
+
+    return { url: null, envVar: err.key }
+  }
+}
+
+async function addDependency (configManager, services, service, dependencyId, urlString) {
+  let { url, envVar } = await parseConfigUrl(configManager, urlString)
+  if (url !== null) {
+    service.dependencies.push({ id: dependencyId, url, local: false })
+    return
+  }
+
+  const dependency = services.get(dependencyId)
+  if (typeof dependency === 'undefined') {
+    throw new errors.MissingDependencyError(
+      missingDependencyErrorMessage(dependencyId, service, services)
+    )
+  }
+
+  url = `http://${dependencyId}.plt.local`
+
+  if (envVar !== null) {
+    service.localServiceEnvVars.set(envVar, url)
+  }
+
+  dependency.dependents.push(service.id)
+  service.dependencies.push({ id: dependencyId, url, local: true })
+}
+
+async function getDependencies (services, service) {
+  const configManager = new ConfigManager({ source: service.config })
```

Comment:
Why are you doing this?

Reply:
I couldn't find a better way to load the service module.
Am I wrong?